README.md

```diff
@@ -2,29 +2,61 @@
 
 [![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](http://standardjs.com/)  [![Build Status](https://travis-ci.org/fastify/fastify-auth.svg?branch=master)](https://travis-ci.org/fastify/fastify-auth)
 
-Fastify authorization - proof of concept.
+This module does not provide an authentication strategy, but it provides a very fast utility to handle authentication (also multiple strategies) in your routes, without adding overhead.  
+Check out the complete example [here](https://github.com/fastify/fastify-auth/blob/master/example.js).
 
-## Usage
-Install the dependencies and run the server:
+## Install
 ```
-npm i
-npm start
+npm i fastify-auth --save
 ```
 
-Then use the following to test the server:
-```
-# route without authorizatrion
-curl  http://localhost:3000/no-auth
+## Usage
+As said above, `fastify-auth` does not provide an authentication strategy, so you must provide it (or them) by yourself, with a decorator or another plugin.
+In the following example you will find a very simple implementation that should help you understand how use this module.  
+```js
+fastify.register(require('fastify-auth'))
 
-# route with authorizatrion
-curl -H "auth: authstring" http://localhost:3000/auth
+fastify.decorate('verifyJWTandLevel', function (request, reply, done) {
+  // your validation logic
+  done() // pass an error if the authentication fails
+})
 
-# register a new user
-curl -H "Content-Type: application/json" -X POST -d '{"user":"abc","password":xyz"}' http://localhost:3000/register
+fastify.decorate('verifyUserAndPassword', function (request, reply, done) {
+  // your validation logic
+  done() // pass an error if the authentication fails
+})
+
+fastify.route({
+  method: 'POST',
+  url: '/auth-multiple',
+  beforeHandler: fastify.auth([
+    fastify.verifyJWTandLevel,
+    fastify.verifyUserAndPassword
+  ]),
+  handler: (req, reply) => {
+    req.log.info('Auth route')
+    reply.send({ hello: 'world' })
+  }
+})
 ```
+`fastify-auth` will run all your authentication methods if at least one succeed it will continue your request, otherwise it wil return an error to the client. It will automatically stop as soon as a authentication succeed.
+
+You can use this plugin on route level, as in the above example or on hook level, by using the `preHandler` hook:
```

Comment:
Can you rephrase a bit, what does 'hook level' means? Is it a plugin-level, via a hook?